src/scripts/exclusion_term_expansion.py

```diff
@@ -0,0 +1,398 @@
+            query_template_path=SPARQL_TERM_REGEX_PATH,
+            regexp=regex_pattern,
+            cache_suffix=f'r{str(index)}')
+        if len(search_result_df) > 0:
+            search_result_df['term_id'] = search_result_df['term_id'].apply(uri_to_curie)
+            for fld in exclusion_fields:
+                search_result_df[fld] = row[fld]
+            search_results.append(search_result_df)
+    if search_results:
+        # - Combine each of the regex search results
+        df_regex_pre_results = pd.concat(search_results)
+        # - Query for children
+        df_regex_results = expand_ontological_exclusions(onto_path, df_regex_pre_results, cache_suffix='2')
+
+    # Combine & massage
+    # - CONCAT: Combine results of all queries
+    df_results = pd.DataFrame()
+    if any([len(x) > 0 for x in [df_explicit_results, df_regex_results]]):
+        df_results = pd.concat([x for x in [df_explicit_results, df_regex_results] if x is not None])
+    # - Drop duplicates: As there is likely to be overlap between (a) explicitly listed terms, and (b) Regex-selected
+    # terms. It is also theoretically possible that there may be some cases where there is a different exclusion_reason
+    # for a term that is within both (a) and (b). In that case, duplicate terms will remain, showing both reasons.
+    df_results = df_results.drop_duplicates()
+
+    return df_results
+
+
+def run(
+    onto_name: str, onto_path: str, exclusions_path: str, mirror_signature_path: str, component_signature_path: str,
+    config_path: str, save=CONFIG['save']
+) -> Union[Dict[str, pd.DataFrame], None]:
+    """Run"""
+    # Vars
+    df_results_path = os.path.join(CONFIG_DIR, f'{onto_name}_exclusion_reasons.robot.template.tsv')
+    df_results_simple_path = os.path.join(CONFIG_DIR, f'{onto_name}_term_exclusions.txt')
+
+    # Prefixes
+    with open(config_path, 'r') as stream:
+        onto_config = yaml.safe_load(stream)
+    prefix_uri_map = onto_config['base_prefix_map']
+    prefix_sparql_strings = [f'prefix {k}: <{v}>' for k, v in prefix_uri_map.items()]
+
+    # Get excluded terms
+    expanded_intensional_exclusions_df = expand_intensional_exclusions(
+        onto_path=onto_path, exclusions_path=exclusions_path, prefix_sparql_strings=prefix_sparql_strings)
+    non_inclusions_df = get_non_inclusions(
+        mirror_signature_path=mirror_signature_path, component_signature_path=component_signature_path,
+        prefix_uris=list(prefix_uri_map.values()))
+    df_results = pd.concat([expanded_intensional_exclusions_df, non_inclusions_df])
+    if len(df_results) == 0:
+        df_results = pd.DataFrame()
+        df_results['term_id'] = ''
+
+    # Add simplified version
+    # - df_results_simple: Simpler dataframe which is easier to use downstream for other purposes
+    df_results_simple = df_results[['term_id']]
+    # - df_results: Add special meta rows
+    # todo: Utilize `has_exclusion_reason`, pending mondo#5177 completion:
```

Comment:
@matentzn I see that Nicole has finished this [`has_exclusion_reason` issue](https://github.com/monarch-initiative/mondo/pull/5187/files) ([pr](https://github.com/monarch-initiative/mondo/issues/5177)).

I'm wondering which of the following I may / may not need to do:
- [ ] 1. Update output row `{'term_id': 'ID', 'exclusion_reason': 'AI rdfs:seeAlso'}` to say `has_exclusion_reason`.
- [ ] 2. Update any inputs where I'm reading from a column called `exclusion_reason` to `has_exclusion_reason`
- [ ] 3. Update the rest of my outputs to use `has_exclusion_reason` instead of `exclusion_reason`

Reply:
1. Great catch! {'term_id': 'ID', 'exclusion_reason': 'AI rdfs:seeAlso'} -> {'term_id': 'ID', 'exclusion_reason': 'AI MONDOREL: has_exclusion_reason'}. You don't have to concern yourself with the `MONDOREL` prefix right now.
2. No, don't worry. No other changes are required. Leave it at `exclusion_reason` or whatever else you have it. Only the ROBOT template above needs to be adjusted!
3. No need either. Leave it as it is!